core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/stopstartds/StopStartDownscaleActions.java

```diff
@@ -91,15 +92,34 @@ protected void prepareExecution(StopStartDownscaleTriggerEvent payload, Map<Obje
             }
 
             @Override
-            protected void doExecute(
-                    StopStartDownscaleContext context, StopStartDownscaleTriggerEvent payload, Map<Object, Object> variables) throws Exception {
-                stopStartDownscaleFlowService.clusterDownscaleStarted(context.getStack().getId(), payload.getHostGroup(), payload.getHostIds());
-                sendEvent(context);
+            protected void doExecute(StopStartDownscaleContext context, StopStartDownscaleTriggerEvent payload, Map<Object, Object> variables) throws Exception {
+                StackDtoDelegate stack = context.getStack();
+                stopStartDownscaleFlowService.clusterRecoveryCandidatesCollectionStarted(stack.getId(), payload.getHostGroup());
+                List<InstanceMetadataView> allInstanceMetadataInHostGroup =
```

Comment:
Not filtering by hostgroup.

This also filters out ZOMBIE instances ... any idea what those are?